numpy/ma/core.py

```diff
@@ -5188,6 +5188,18 @@ def mean(self, axis=None, dtype=None, out=None, keepdims=np._NoValue):
         """
         kwargs = {} if keepdims is np._NoValue else {'keepdims': keepdims}
 
+        is_float16_result = False
+
+        # Cast bool, unsigned int, and int to float64 by default
+        if dtype is None:
+            if issubclass(self.dtype.type, (ntypes.integer, ntypes.bool_)):
+                dtype = ntypes.dtype('f8')
```

Comment:
I think the right side can be written as just `np.float64`.  Technically, that is the scalar type and not a `dtype` instance, but it should work.

Reply:
I got inspired / copied this from the implementation at https://github.com/numpy/numpy/blob/master/numpy/core/_methods.py#L142-L147, but I will change the things as you suggest.